dns/data_dns_mx_record_set.go

```diff
@@ -0,0 +1,73 @@
+package dns
+
+import (
+	"fmt"
+	"net"
+	"sort"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceDnsMXRecordSet() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceDnsMXRecordSetRead,
+		Schema: map[string]*schema.Schema{
+			"domain": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"mx": &schema.Schema{
+				Type: schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"preference": {
+							Type:     schema.TypeInt,
+							Required: true,
```

Comment:
Should this be computed?

Reply:
I don't think so, no. TBH I can't see any mention of `Computed` in any of the other DNS resources.

Reply:
There are no other data sources with nested attributes in the DNS provider IIRC. In the azure providers [we mark all parts](https://github.com/terraform-providers/terraform-provider-azurerm/blob/623b2920b992784c60ecbb20f7c404314c06533d/azurerm/data_source_kubernetes_cluster.go#L27) of a sub block as computed. Required makes little sense he as the user will never set them.